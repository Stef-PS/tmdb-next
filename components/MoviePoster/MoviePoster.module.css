.MoviePoster {
  width: 100%;
}

.MoviePoster img {
  width: 100%;
  height: 100%;
  object-fit: cover;
}
