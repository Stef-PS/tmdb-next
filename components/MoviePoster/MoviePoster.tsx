import { useState } from 'react'
import styles from './MoviePoster.module.css' 

interface MoviePosterProps {
  imageUrl: string,
  fallbackUrl: string,
  alt: string,
  aspectRatio?: string,
  testError?: boolean
}

const MoviePoster = ({ imageUrl, fallbackUrl, alt, aspectRatio = '', testError = false }: MoviePosterProps) => {
  const [error, setError] = useState(testError)
  const errorCallback = () => fallbackUrl && setError(true)
  return (
    <div className={styles.MoviePoster} style={{ aspectRatio }}>
      <img src={error ? fallbackUrl : imageUrl} alt={alt} onError={errorCallback} />
    </div>
  )
}

export default MoviePoster
