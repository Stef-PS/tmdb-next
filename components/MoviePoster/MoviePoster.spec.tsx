import { render } from '@testing-library/react'
import MoviePoster from './MoviePoster'
import '@testing-library/jest-dom'

const validImageUrl = 'https://via.placeholder.com/300/00FF00/000000?text=valid'
const fallbackImageUrl = 'https://via.placeholder.com/300/FF0000/FFFFFF?text=fallback'

describe('MoviePoster', () => {
  it('should render as usual', () => {
    const { container } = render(<MoviePoster imageUrl={validImageUrl} fallbackUrl={fallbackImageUrl} alt="movie poster" />)
    expect(container).toMatchSnapshot()
  })

  it('should render an img tag with the right src', () => {
    const { getByAltText } = render(<MoviePoster imageUrl={validImageUrl} fallbackUrl={fallbackImageUrl} alt="movie poster" />)
    const img = getByAltText('movie poster')
    expect(img).toBeInTheDocument()
    expect(img).toHaveAttribute('src', validImageUrl)
  })

  it('should render an img tag with the fallback src', () => {
    const { getByAltText } = render(<MoviePoster imageUrl={validImageUrl} fallbackUrl={fallbackImageUrl} alt="movie poster" testError={true} />)
    const img = getByAltText('movie poster')
    expect(img).toBeInTheDocument()
    expect(img).toHaveAttribute('src', fallbackImageUrl)
  })
})
