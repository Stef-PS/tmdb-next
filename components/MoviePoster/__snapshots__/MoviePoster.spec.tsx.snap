// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviePoster should render as usual 1`] = `
<div>
  <div
    class="MoviePoster"
  >
    <img
      alt="movie poster"
      src="https://via.placeholder.com/300/00FF00/000000?text=valid"
    />
  </div>
</div>
`;
